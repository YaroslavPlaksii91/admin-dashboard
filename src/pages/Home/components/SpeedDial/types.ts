import { MouseEventHandler } from 'react';

export type SpeedDialProps = {
  handleDelete: MouseEventHandler<HTMLDivElement>;
  handleEdit: MouseEventHandler<HTMLDivElement>;
};
