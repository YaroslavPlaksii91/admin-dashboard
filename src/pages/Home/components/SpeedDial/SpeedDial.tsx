import { FC } from 'react';
import {
  SpeedDial as SpeedDialMui,
  SpeedDialAction,
  SvgIcon,
} from '@mui/material';
import EditIcon from '@mui/icons-material/Edit';
import DeleteIcon from '@mui/icons-material/Delete';

import { SpeedDialProps } from './types';

export const SpeedDial: FC<SpeedDialProps> = ({ handleDelete, handleEdit }) => {
  const actions = [
    { icon: <DeleteIcon />, name: 'Delete', onClick: handleDelete },
    { icon: <EditIcon />, name: 'Edit', onClick: handleEdit },
  ];

  return (
    <SpeedDialMui
      ariaLabel="SpeedDial more options"
      sx={{ height: 24 }}
      icon={
        <SvgIcon titleAccess="More options">
          <use href="/sprite.svg#icon-more"></use>
        </SvgIcon>
      }
    >
      {actions.map(action => (
        <SpeedDialAction
          key={action.name}
          icon={action.icon}
          tooltipTitle={action.name}
          sx={{ height: 24 }}
          onClick={action.onClick}
        />
      ))}
    </SpeedDialMui>
  );
};
