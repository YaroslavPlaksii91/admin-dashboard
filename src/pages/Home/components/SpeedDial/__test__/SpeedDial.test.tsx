import { cleanup, render, screen, fireEvent } from '@testing-library/react';

import { SpeedDial } from '../SpeedDial';

describe('SpeedDial component', () => {
  afterEach(cleanup);

  const handleDelete = jest.fn();
  const handleEdit = jest.fn();

  it('should be rendered', () => {
    const { asFragment } = render(
      <SpeedDial handleDelete={handleDelete} handleEdit={handleEdit} />,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly with actions', () => {
    render(<SpeedDial handleDelete={handleDelete} handleEdit={handleEdit} />);

    const moreOptionsIcon = screen.getByTitle('More options');
    expect(moreOptionsIcon).toBeInTheDocument();

    const actionButtons = screen.getAllByRole('menuitem');

    actionButtons.forEach(button => expect(button).toBeInTheDocument());
  });

  it('calls handleDelete when Delete action is clicked', () => {
    render(<SpeedDial handleDelete={handleDelete} handleEdit={handleEdit} />);

    const actionButtons = screen.getAllByRole('menuitem');

    fireEvent.click(actionButtons[0]);

    expect(handleDelete).toHaveBeenCalledTimes(1);
    expect(handleEdit).not.toHaveBeenCalled();
  });

  it('calls handleEdit when Edit action is clicked', () => {
    const handleDelete = jest.fn();
    const handleEdit = jest.fn();

    render(<SpeedDial handleDelete={handleDelete} handleEdit={handleEdit} />);

    const actionButtons = screen.getAllByRole('menuitem');

    fireEvent.click(actionButtons[1]);

    expect(handleEdit).toHaveBeenCalledTimes(1);
    expect(handleDelete).not.toHaveBeenCalled();
  });
});
