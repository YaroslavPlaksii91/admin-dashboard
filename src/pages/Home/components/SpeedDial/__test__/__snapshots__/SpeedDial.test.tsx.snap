// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpeedDial component should be rendered 1`] = `
<DocumentFragment>
  <div
    class="MuiSpeedDial-root MuiSpeedDial-directionUp css-5jgzwq-MuiSpeedDial-root"
    role="presentation"
  >
    <button
      aria-controls="SpeedDialmoreoptions-actions"
      aria-expanded="false"
      aria-haspopup="true"
      aria-label="SpeedDial more options"
      class="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeLarge MuiFab-primary MuiFab-root MuiFab-circular MuiFab-sizeLarge MuiFab-primary MuiSpeedDial-fab css-7dv1rb-MuiButtonBase-root-MuiFab-root-MuiSpeedDial-fab"
      style="transform: none; webkit-transition: transform 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: transform 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="0"
      type="button"
    >
      <svg
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        focusable="false"
        role="img"
        viewBox="0 0 24 24"
      >
        <use
          href="/src/assets/icons/sprite.svg#icon-more"
        />
        <title>
          More options
        </title>
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <div
      aria-orientation="vertical"
      class="MuiSpeedDial-actions MuiSpeedDial-actionsClosed css-1s28u9o-MuiSpeedDial-actions"
      id="SpeedDialmoreoptions-actions"
      role="menu"
    >
      <button
        aria-label="Delete"
        class="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default MuiSpeedDialAction-fab MuiSpeedDialAction-fabClosed css-tmh29y-MuiButtonBase-root-MuiFab-root-MuiSpeedDialAction-fab"
        data-mui-internal-clone-element="true"
        role="menuitem"
        style="transition-delay: 60ms;"
        tabindex="-1"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="DeleteIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        aria-label="Edit"
        class="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default MuiSpeedDialAction-fab MuiSpeedDialAction-fabClosed css-tmh29y-MuiButtonBase-root-MuiFab-root-MuiSpeedDialAction-fab"
        data-mui-internal-clone-element="true"
        role="menuitem"
        style="transition-delay: 30ms;"
        tabindex="-1"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="EditIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
